phidgets_high_speed_encoder/src/phidgets_high_speed_encoder.cpp

```diff
@@ -0,0 +1,250 @@
+  ROS_INFO("Version: %d", version);
+  ROS_INFO("Number of encoders %d", num_encoders);
+
+  return 0;
+}
+
+bool attach(
+    CPhidgetEncoderHandle &phid,
+    int serial_number)
+{
+  // create the object
+  CPhidgetEncoder_create(&phid);
+
+  // Set the handlers to be run when the device is
+  // plugged in or opened from software, unplugged or
+  // closed from software, or generates an error.
+  CPhidget_set_OnAttach_Handler((CPhidgetHandle) phid,
+                                AttachHandler, NULL);
+  CPhidget_set_OnDetach_Handler((CPhidgetHandle) phid,
+                                DetachHandler, NULL);
+  CPhidget_set_OnError_Handler((CPhidgetHandle) phid,
+                               ErrorHandler, NULL);
+
+  // Registers a callback that will run if an input changes.
+  // Requires the handle for the Phidget, the function
+  // that will be called, and an arbitrary pointer that
+  // will be supplied to the callback function (may be NULL).
+  CPhidgetEncoder_set_OnInputChange_Handler(phid,
+                                            InputChangeHandler,
+                                            NULL);
+
+  // Registers a callback that will run if the
+  // encoder changes.
+  // Requires the handle for the Encoder, the function
+  // that will be called, and an arbitrary pointer that
+  // will be supplied to the callback function (may be NULL).
+  CPhidgetEncoder_set_OnPositionChange_Handler(phid,
+                                               PositionChangeHandler,
+                                               NULL);
+
+  //open the device for connections
+  CPhidget_open((CPhidgetHandle) phid, serial_number);
+
+  // get the program to wait for an encoder device
+  // to be attached
+  if (serial_number == -1)
+  {
+    ROS_INFO("Waiting for High Speed Encoder Phidget " \
+         "to be attached....");
+  }
+  else
+  {
+    ROS_INFO("Waiting for High Speed Encoder Phidget " \
+         "%d to be attached....", serial_number);
+  }
+  int result;
+  if ((result =
+           CPhidget_waitForAttachment((CPhidgetHandle) phid,
+                                      10000)))
+  {
+    const char *err;
+    CPhidget_getErrorDescription(result, &err);
+    ROS_ERROR("Problem waiting for attachment: %s",
+              err);
+    return false;
+  }
+  else return true;
+}
+
+/*!
+ * \brief disconnect the encoder
+ */
+void disconnect(CPhidgetEncoderHandle &phid)
+{
+  ROS_INFO("Closing...");
+  CPhidget_close((CPhidgetHandle) phid);
+  CPhidget_delete((CPhidgetHandle) phid);
+}
+
+int main(int argc, char *argv[])
+{
+  ros::init(argc, argv, "phidgets_high_speed_encoder");
+  ros::NodeHandle n;
+  ros::NodeHandle nh("~");
+  int serial_number = -1;
+  nh.getParam("serial", serial_number);
+  std::string name = "encoder";
+  nh.getParam("name", name);
+  if (serial_number == -1)
+  {
+    nh.getParam("serial_number", serial_number);
+  }
+  ROS_INFO("frame_id = %s", frame_id.c_str());
```

Comment:
This prints the `frame_id` before it is read from the parameter server (line 222).